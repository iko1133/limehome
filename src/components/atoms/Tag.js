import React from "react";
import styled from "styled-components/native";

const TagContainer = styled.View`
  background-color: ${({ theme }) => theme.TAG_BACKGROUND_COLOR};
  border-radius: 6px;
  padding: 4px;
`;

const TagText = styled.Text`
  color: ${({ theme }) => theme.TAG_TEXT_COLOR};
`;

const Tag = ({ text, style }) => {
  return (
    <TagContainer style={style}>
      <TagText>{text}</TagText>
    </TagContainer>
  );
};

export default Tag;
