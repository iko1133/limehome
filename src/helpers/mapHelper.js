// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
