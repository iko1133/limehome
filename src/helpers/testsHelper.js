import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
