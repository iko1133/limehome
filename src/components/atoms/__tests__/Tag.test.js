import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import Tag from "../Tag";

describe("Testing Tag", () => {
  it("Renders Tag with 'Small tag' text in it", () =>
    expect(
      renderer.create(WrapInThemeProvider(<Tag text="Small tag" />))
    ).toMatchSnapshot());

  it("Renders Tag with 'THIS IS A BIG, LONG, HUGE TAG' text in it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(<Tag text="THIS IS A BIG, LONG, HUGE TAG" />)
      )
    ).toMatchSnapshot());
});
