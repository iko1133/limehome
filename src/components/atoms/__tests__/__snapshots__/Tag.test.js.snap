// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tag Renders Tag with 'Small tag' text in it 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CACBAB",
        "borderRadius": 6,
        "paddingBottom": 4,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#7A7B70",
        },
      ]
    }
  >
    Small tag
  </Text>
</View>
`;

exports[`Testing Tag Renders Tag with 'THIS IS A BIG, LONG, HUGE TAG' text in it 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CACBAB",
        "borderRadius": 6,
        "paddingBottom": 4,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 4,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#7A7B70",
        },
      ]
    }
  >
    THIS IS A BIG, LONG, HUGE TAG
  </Text>
</View>
`;
